ingest/api/model.py

```diff
@@ -339,6 +331,23 @@ def validate_wigos_id(self) -> Properties:
 
         return self
 
+    @model_validator(mode="after")
+    def transform_period_to_seconds(self):
+        try:
+            duration = isodate.parse_duration(self.period)
+        except isodate.duration.ParsingError:
+            raise ValueError("Invalid duration format.")
+
+        if isinstance(duration, timedelta):
+            # It's a simple timedelta, so just get the total seconds
+            total_seconds = duration.total_seconds()
+
+        else:
+            raise ValueError("Duration not convertable to seconds.")
```

Comment:
That check is necessary . But a correction is needed. `if isinstance(duration, isodate.duration.Duration)`

Reply:
@fjugipe Think this also involves your functions. Any comments?

Reply:
In `api.utilities.py` it is `if isinstance(duration, isodate.duration.Duration)` and it was originally used just to be consistent with `ingest` to convert years and months as it previously had some special conversions. But I don't think that it is needed there if we are going to use the duration parsing of `isodate` library in `ingest`. It should be consistent as is.

But! In model validation I think the check is good, since `parse_duration` returns either a `timedelta` or a `Duration` object. `Duration`  object supports months and years, but `timedelta` does not. Both of them are convertable to seconds.

So if we decide to support only max period of `P1D` this is the place, where we can just drop the date if it is not converted to a `timedelta`. But if we want to support for example `P1M` we can use `if isinstance(duration, isodate.duration.Duration) or isinstance(duration, timedelta)` and drop the date if it is not convertable to seconds.

Reply:
We disuccused this at KNMI, and we will just not support anything that is not consistently convertable to seconds, so `M` are not allowed. Therefore is the conversion returns anything other then timedelta, the ingest is rejected.